### DISCUSSION TITLE
Best approach for chip size other than 10 x 10 mm?

### DISCUSSION THREAD

**Discussion by zachparrott** [iqm-finland/KQCircuits#65] [Q&A] [answered]
Hi. I have been enjoying incorporating KQCircuits in my workflow, but not sure best way to proceed with writing chip classes for designs for sizes other than 10 x 10 mm. I know in the `produce_n_launchers` method of the `Chip` class I can pass the `chip_box` parameter but wasn't sure if there was another preferred way?

**Comment by qpavsmi** [comment, marked as answer]
Hi, I'm so glad to hear that you are using KQCircuits!

The preferred way to set the dimensions of your chip is to add the following class decorator before your chip class definition:

```
@add_parameters_from(ChipFrame,
                     box=pya.DBox(pya.DPoint(0, 0), pya.DPoint(20000, 20000)))
class YourChip(Chip):
...
```
Set the `DBox` dimensions to your liking, and it doesn't need to be a perfect square.
If you are considering to create a flip chip, to set the chip dimensions of other chips in the stack, add another decorator like this:
```
@add_parameters_from(Chip,
                     face_boxes=[None, pya.DBox(pya.DPoint(1500, 1500), pya.DPoint(18500, 18500))])
```
The first element `None` will define the first chip to use the dimensions defined in `box` parameter, and the next `DBox` element will set dimensions of the second chip on top.

Make sure to check what other parameters need changing after changing the chip dimensions, for example `frames_marker_dist`.

Hope this helps you started and let me know if this causes other issues.

**Reply by zachparrott** [reply]
Thanks for the quick reply and help Pavel!
